### ISSUE TITLE
Add Contributors/Maintainers To Readme's

### ISSUE THREAD

**Issue by pathartl** [TheOtherGuys-Hakchi-Projects/Hakchi_RetroArch_Cores#3]
The current readme's only contain information about the creator for each core. This should be designated to the original creator of the core which in most cases is the libretro team. Anyone that successfully gets running on the Nintendo Classic systems should be credited as maintainers/contributors as to correctly preserve credit where credit is due.

**Comment by pathartl** [comment]
Ref PR #2 
This is exactly the thing I think we should avoid. It also would provide a better expectation of a contributor's role in the core. i.e. most core creators did not write the code backing up the actual functionality of the core. If an issue arises with the core, users might seek out support or blame the wrong entity.

**Comment by BananaRama203** [comment]
@KMFDManic

**Comment by KMFDManic** [comment]
The Author, as DanTheMan827, and myself, and the rest of TeamShinkansen have discussed, in the past, is the person who Authored the HMODS, It is a very specific platform.  Core by libretro is clearly in each HMOD.  But, they had zero to do with the actual HMOD, or how it effectively works on the Mini NES/SNES, the Author does.  HMODS are a Mini NES/SNES thing, not a libretro thing.  Someone makes an engine for a Car, are they the one that has their name on the Car?  No, it is the person who put the engine in the Car:)  Libretro people are all great.  But, it is absolute disrespect to take away Author Credit of those who got these things running on the Minis.  It took an incredulous amount of effort, work, patience, and time, communication and more.  This is a small thing to change, where Credit is due.

**Comment by pathartl** [comment]
@KMFDManic it took even more time to write the actual cores and emulate the actual platform that the games were developed for. I am by no means saying that credit should not be given to the maintainer of the HMOD. However, we do need to maintain perspective.

Without the libretro/retroarch team we would effectively be nowhere in this project. Credit needs to be given properly.

Ideally, what our repository should contain is submodules to a repository for each core. Each core would contain the necessary build configuration required for the Classics and each core repository would contain a submodule for the libretro repository. For example, this repository would have a reference to the fbalpha core repository. That repository would contain necessary scripts to build/package the HMOD. The repository would also have a submodule pointing to [the source of the libretro fbalpha core](https://github.com/libretro/fbalpha). This makes building 100% easier and establishes a scalable/maintainable/consistent development process.

Besides, we shouldn't keep binaries in repositories. I don't think this repo should be a distribution channel.

**Comment by pathartl** [comment]
Then commit the source, credit libretro properly, and contribute/share with others how you were able to tackle the issues you ran into. It is disrespectful to the community to act as the sole proprietor.

Again, I am not trying to suggest to take credit away from anyone. I do not want credit or information lost.

**Comment by pathartl** [comment]
Any other core maintainers want to share their mind?

**Comment by KMFDManic** [comment]
NOTE: the libretro commits portion, given your statement to credit properly.  This is something I have already been working on for ALL Cores.

I don't want singular libretro as who did the Core, which is why I have been implementing each and every individual who was behind the commits for any of these given Cores.  It is a painstaking, very time consuming task.  But, I have been slowly, but surely been adding them.  I am the Author of the HMOD.  libretro as Credit, does not give proper credit to the true people behind the commits.  This is why I have already gone out of my way to put the additional section for libretro commits.  And, I plan on having them all more fleshed out, in the near future.  Open Source means you can freely work with and distribute, as long as the original people are credited.  I have made it an absolute point to thank many of these people personally.  In fact, several follow my work and have thanked me as well for helping get their Cores on our Minis.  And, acting as sole proprietor is not what I am doing here.  I created the HMOD, and put my name as the creator of the HMOD.  I properly credited libretro as we have already been doing since NESC Days.  This system has already been in place, with noone having a problem with it, for nearly 2 years now.  Credit is properly given to the Creator of the HMOD, as well as libretro, the developer organization the Core comes from, as well as whom made commits for said Cores, within libretro. 

Name: BlueMSX 
Creator: KMFDManic
Category: RetroArch Cores
-----------------------
=== BlueMSX Core for RetroArch ===

Port of blueMSX to the libretro API

Module adds support for Coleco Vision & MSX

Bios must be inside game directory or via Bios Module where applicable

Available executables and arguments to run Core:
- /bin/col <rom> <clover_args>
- /bin/msx <rom> <clover_args>
- /bin/bluemsx <rom> <clover_args>

Core by libretro 

libretro commits by:
Alcaro/aliaspider/bparker06/cxx/fr500/garbear/
GregorR/iKarith/meepingsnesroms/msheehan79/
Oggom/r-type/sergiobenrocha2/Tatsuya79/
twinaphex/vanfanel/yoshisuga

Hakchi module system by madmonkey

NES/SNES Mini shell integration by Cluster

(c) 2016-2018

**Comment by KMFDManic** [comment]
-----------------------
Name: PPSSPP 
Creator: KMFDManic/madmonkey/'TheOtherGuys'
Category: RetroArch Cores
-----------------------

I can also change to be this.  And, as people collaborate and changes are made, they get added to the Credits, such as have gpstar, madmonkey, pcm, and a few others, already.  libretro credits can be in the 100-200 people range, for any given Core.  I am just trying to keep with appropriate Credits, as we have already established long ago:)  The libretro commits by: section, was my own personal addition, as I felt libretro as a whole..truly didn't thank the many individuals involved.  This is why I have really personally thanked over 100 of them, because of things they have done for the scene.

As far as binaries, going for source, instead...would create double work.  You'd need to compile, then have it package to HMOD form.  The key things changed, given the binaries are already done and compiled, would be the launcher scripts, and so on.  We do not operate on quite the same level as the PC front, as they do not use HMODS.  This is a Mini NES/SNES specific thing.

**Comment by pathartl** [comment]
Can't respond to everything right now, but: 

> As far as binaries, going for source, instead...would create double work. You'd need to compile, then have it package to HMOD form. The key things changed, given the binaries are already done and compiled, would be the launcher scripts, and so on. We do not operate on quite the same level as the PC front, as they do not use HMODS. This is a Mini NES/SNES specific thing.

That is not double work. That is how it should be done. Git repositories are not distribution channels.

**Comment by BananaRama203** [comment]
Hi all, sorry I have been away. I believe Melthris has been is talks directly with the libretro dev team and they have voiced their own opinions abouts the readmes. **Which means we must be respectful of and abide by their wishes.**

I believe the solution is to have the creators as Libretro Team + Individual core developer(s) and then have the "Hakchi" side of credits near the bottom as it is now with KMFD/MM/Cluster but maybe put it in bold?

Melthris is doing a draft and getting it approved by Libretro and once it's approved we will roll with that.

**Comment by KMFDManic** [comment]
@swingflip, this means 'TheOtherGuys' are no longer Creator of the NEO RetroArch HMOD, nor Advokaten of the XMB HMOD, as well as many other conflicts, given HMOD Creator had always heen original intent for this entry...so that Hakchi2 CE can separate by HMOD Creator.  This will make it impossible for CE to separate by HMOD Creator if all say libretro.  Libretro needs to stay "Core created by libretro", so proper separation can be utilized in future CE planned Updates. If this an issue, I will change ReadMes to better reflect whom did the HMODS originally, on my end.

**Comment by BananaRama203** [comment]
We can/will review the Retroarch thing as you raise a good point. We need to ensure proper credit is given for the mods. RetroArch as a whole is a bit different than the cores as it is a lot more involved compared to the cores for example. There is a lot of reworking and coding involved with the RA which you can class as it's own forked project and is a decent  amount different from the originating project.

As stated before Melthris should have the template which Libretro are happy with and I will get him to review the readme for the RA too.

**Comment by KMFDManic** [comment]
@swingflip The only possible way to do this is modify all ReadMes to have libretro shared with Creator credit.  You have no way of knowing how much went into a Core versus Retroarch.  Not at all an argument to leave 'TheOtherGuys', versus following a standard shared Credit practice.

Creator: libretro, 'TheOtherGuys'

Creator: libretro, KMFDManic

Shared credit is the only possible way to guarantee CE can still separate by Creator, and appease those involved in the process.  Not, who did more or less.  This is something that could never properly be determined.

**Comment by Melthris** [comment]
Hi guys.

After many discussions with most of you and a few others, I would like to propose a format that everyone needs to agree on in regards to display readme's for HMODS and all other work in relation to the miniSNES.

I preface this with three things first;
- I've spoken with some Libretro people in regards to this (they've actually already been aware and in the loop this the older RetroArch work and core work for quite a long time)
- I understand and appreciate where all of you are coming from.
- This post is intended to set a standard for everything

One could contend that if we were to set one standard or set of expectations for one set of work while setting a different standard or set of expectations for another set of work, we would see potential unfair treatment to those people who are contributing on their respective bits of work and overall, this would confuse the community who download these contributions.

So we fix this by setting a standard for everyone.

To KMFDManic's credit, his readme's cover most of this already as do some other's readme files but articulating this standard is what's important.

The format of each readme needs to include;
- The name of the HMOD
- Credit the creator of the HMOD (this isn't where Libretro's name is mentioned or id Software's name is mentioned for say, the DOOM HMOD. This is the creator of the HMOD. The person or the team who compiled everything together and made it HMOD ready)
- Note the category (which is another discussion for another day)
- A description of what the HMOD does. If it's visual, we need to look at providing a screenshot or two
- Include executables/arguments (if we're looking at say Core work)
- Note anything else of importance points of the HMOD
- credits to all appropriate parties and includes anything else of importance.

So the way a HMOD's readme should look, if I was the one who created it would be;

> 
> -----------------------
> Name: My first User Interface HMOD
> Creator: Melthris
> Category: UI modification
> -----------------------
> 
> **Description:**
> This HMOD will allow you to re-arrange the the layout of the 'game tiles' on the miniSNES' interface to and makes them larger than they currently are by 50%.
> 
> **Screenshots**
> PreviewScreenShotHere.png
> 
> ===================
> 
> If you are having issues with this HMOD, please read this **_FAQ_**. Alternatively, if the FAQ was not able to remedy your issues, please visit us on www.reddit.com/r/miniSNESmods or chat to us live on Discord! **DISCORD URL HERE**
> 
> ===================
> 
> **Shoutouts/credits:**
> Hakchi-Resource commits by:
> Swingflip/Patton Plays/Jesus/Earthworm Jim/Blah blah blah
> 
> Hakchi module system by madmonkey
> 
> NES/SNES Mini shell integration by Cluster
> 
> (c) 2016-2018
> 

I want to note that the current structure KMFDManic for his credit's in his footnotes already follows this format and is entirely acceptable for the core's he is working on as well. It hasn't been an issue for almost two years or so, so it shouldn't be an issue now.

It is imperative that everyone agree to this format moving forward. 

Does this work for everyone?

**Comment by BananaRama203** [comment]
Hi everyone, after lengthy discussions with Melthris and running through scenarios I believe the final decision is the following:

> 
> -----------------------
> Name: RetroArch Core X
> Creator: Libretro / KMFDManic
> Category: UI modification
> -----------------------
> 
> **Description:**
> This HMOD will allow you to re-arrange the the layout of the 'game tiles' on the miniSNES' interface to and makes them larger than they currently are by 50%.
> 
> **Screenshots**
> PreviewScreenShotHere.png
> 
> ===================
> 
> If you are having issues with this HMOD, please read this **_FAQ_**. Alternatively, if the FAQ was not able to remedy your issues, please visit us on www.reddit.com/r/miniSNESmods or chat to us live on Discord! **https://discord.gg/8gygsrw**
> 
> ===================
> 
> **Shoutouts/credits:**
> Hakchi-Resource commits by:
> Swingflip/Patton Plays/Jesus/Earthworm Jim/Blah blah blah
> 
> Hakchi module system by madmonkey
> 
> NES/SNES Mini shell integration by Cluster
> 
> (c) 2016-2018
> 

basically we will have it so stuff like doom, RA and the cores will be
DOOM = ID SOFTWARE/Swingflip
RA = Libretro/"TheOtherGuys"
CORES = Libretro/KMFDManic

etc.

I will be updating the doom,quake, RA etc so I believe the final decision is fair and just.

**Comment by Melthris** [comment]
Yeah the above seems to be what is agreed on as being reasonable.

The 'creator' section would essentially cover any progenitor of the work prior to the creator of the HMOD so in DOOM's instance, it's id Software/etc

Same deal would apply if I were to say take inspiration and build off/take elements fromsomeone else's HMOD and include it in my own. Only in this instance, I would let the original creator know prior to it being published via the Mod Store as it's likely going to be someone from HR/miniSNESmods/someone in the community anyway and give them a heads up. They then get the opportunity to officially decline being mentioned in the credits or confirm that they're okay with being included in the credits.

I mean I don't know ANYONE that wouldn't want to be created for their work or someones work that was built off of theirs but I think we should agree on this format as well just in case there is that odd case were this to happen.

**Comment by KMFDManic** [comment]
@swingflip @Melthris The original Authors of the content that libretro and ourselves have open sourced are already covered in the info files.  So, this additional step is not necessary.  Why overly complicate things, when...essentially, an issue is being made out of a non-issue.  The way it is currently set up is properly addressed, as far as original content we open sourced from (within info files).  This is how the licensing agreements are set up for them to begin with, with 3DO as an example.

ie: 

4DO

authors = "JohnnyDude|FreeDO team"
manufacturer = "Various"

PrBoom

authors = "Florian Schulze"
manufacturer = "Id Software"

PPSSPP

authors = "Henrik Hrydgard"
manufacturer = "Sony"

Creator of the HMOD should only contain those who have worked on the HMOD, libretro should be listed as Core by libretro, along with libretro commits by (below), as I have established in this BlueMSX ReadMe.  And, of course, Category, can be determined by how each particular set is done.

-----------------------
Name: BlueMSX
Creator: KMFDManic
Category: RetroArch Cores
-----------------------
=== BlueMSX Core for RetroArch ===

Port of blueMSX to the libretro API

Module adds support for Coleco Vision & MSX

Bios must be inside game directory or via Bios Module where applicable

Available executables and arguments to run Core:
- /bin/col <rom> <clover_args>
- /bin/msx <rom> <clover_args>
- /bin/bluemsx <rom> <clover_args>

Core by libretro 

libretro commits by:
Alcaro/aliaspider/bparker06/cxx/fr500/garbear/
GregorR/iKarith/meepingsnesroms/msheehan79/
Oggom/r-type/sergiobenrocha2/Tatsuya79/
twinaphex/vanfanel/yoshisuga

Hakchi module system by madmonkey

NES/SNES Mini shell integration by Cluster

(c) 2016-2018


In effect, to change Creator to add what is already included in the info files, would need not only Libretro, but also the original author of the PC Emulator, as well as the manufacturer of the original System that the Emulator is based upon.  This overly complicates things.  For 3DO, you'd have to list each and every manufacturer that made the System, to follow this methodology, of which there were several.  3DO made did a goofy thing and licensed out the System to a dozen companies, essentially making them all "Creators":)  This is why it is established as "Various" for manufacture due to the sheer number of them that took on the license.

I have already followed the proper legal documentation as far as setting things up for BlueMSX in particular.

Example license I followed, when doing these:)

https://github.com/libretro/FreeIntv/blob/master/LICENSE

That being said, original Author is 

authors = "Daniel Vik"

And, being that over 100 companies manufactured MSX Computers!

manufacturer = "Various"

And, of course, dictation of the actual license followed in setting things up

license = "GPLv2"

Whom set this standard up?  Well, libretro did:)  I am doing it exactly as they have already been doing it for years:)  Not to mention, this info information is clearly displayed within RetroArch, under Core Information:)
